File: articles/ai-agents-tool-use-and-procedural-autonomy/go/agent_permission.go
package main

import "fmt"

func main() {
	risk := 0.85
	approvalRequired := true
	approved := false
	status := "pass"
	if approvalRequired && !approved {
		status = "blocked"
	} else if risk >= 0.65 {
		status = "escalate"
	}
	fmt.Printf("agent_action_status=%s\n", status)
}

File: articles/ai-agents-tool-use-and-procedural-autonomy/haskell/AgentPermission.hs
module Main where

status :: Double -> Bool -> Bool -> String
status risk approvalRequired approved
  | approvalRequired && not approved = "blocked"
  | risk >= 0.65 = "escalate"
  | otherwise = "pass"

main :: IO ()
main = putStrLn ("agent_action_status=" ++ status 0.85 True False)
